homework10/configuration/FileLoggerConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package homework10.configuration;
+
+import homework10.enums.LoggingLevel;
+
+import java.util.Objects;
+
+public class FileLoggerConfiguration {
+    private String filename;
+    private LoggingLevel level;
+    private long maxSize;
+    private String format;
+
+    public FileLoggerConfiguration(String filename, LoggingLevel level, long maxSize, String format) {
+        this.filename = Objects.requireNonNull(filename, "Filename cannot be null");
+        this.level = Objects.requireNonNull(level, "Logging level cannot be null");
+        this.maxSize = Math.max(Objects.requireNonNull(maxSize, "Max size cannot be null"), 0);

```

Comment:
`maxSize` is a primitive, so you should not check it via `Objects.requireNonNull(maxSize)`, so remove this check. Create a separate method and do this check within it and if the check is passed, return `maxSize`, so it can be assigned in the same line in the constructor.